api/net/Controllers/AuthController.cs

```diff
@@ -59,12 +54,28 @@ public AuthController(ICssHelper cssHelper, IUserService userService, IOptions<J
     /// <returns></returns>
     [HttpPost("userinfo")]
     [Produces(MediaTypeNames.Application.Json)]
-    [ProducesResponseType(typeof(PrincipalModel), 200)]
+    [ProducesResponseType(typeof(PrincipalModel), (int)HttpStatusCode.OK)]
     [SwaggerOperation(Tags = new[] { "Auth" })]
-    public async Task<IActionResult> UserInfoAsync()
+    public async Task<IActionResult> UserInfoAsync(LocationModel? location)
```

Comment:
Optionally we can supply the user location.  
Later I'll extract the user IP from the request to the API to ensure it isn't blocked.